src/features/courses/course-controller.ts

```diff
@@ -2357,19 +2363,33 @@ class CourseController {
     async getQuestions(options: GetQuestionsOptions): Promise<CourseWWTopicQuestion[]> {
         const {
             courseTopicContentId,
-            userId
+            userId,
+            studentTopicAssessmentInfoId,
         } = options;
 
         try {
             const include: sequelize.IncludeOptions[] = [];
+            const subInclude: sequelize.IncludeOptions[] = [];
+
+            if (!_.isNil(studentTopicAssessmentInfoId)) {
+                subInclude.push({
+                    model: StudentGradeInstance,
+                    as: 'gradeInstances',
+                    required: true,
```

Comment:
Why is this required true?
If we missing records does it make sense to return them and then log errors with null checks

Reply:
if the request includes a specific student_topic_assessment_info_id, then there *must* be associated student grade instances. That ID identifies a specific exam version.

Reply:
So you are saying that:
since student topic assessment exists then grade instances must exist

I can agree with that, however, we probably need the nil check anyway and if the generation has a mistake then we will catch it in those nil checks

With required true it would return that student the student grade instance didn't exist and then we would have a red herring of a bug

Reply:
I am saying that the student-topic-assessment-info and its corresponding student-grade-instance children are created in the same transaction. You cannot have either independently.

So, the recommendation is a nil check on the result of this query before returning?

Reply:
not before returning, the type has it as an optional field so when you go to access it you would have to nil check there anyway, I'm saying we have one of our classic impossible errors be thrown at that point

Although I agree that currently, it is not possible with the transaction all it takes is either:
* for someone to delete one of the awaits by accident and have it as an orphaned promise thus not respecting the transaction (starting a new one)
* for some change to come in that someone creates these objects independently and do it incorrect